node/rpc/src/chain_api.rs

```diff
@@ -150,13 +150,22 @@ where
 /// Prints hex-encoded representation of SHA-256 checksum and saves it to a file
 /// with the same name but with a `.sha256sum` extension.
 async fn save_checksum(source: &Path, hash: Output<Sha256>) -> Result<()> {
-    let encoded_hash = format!("{} {}", hash.encode_hex::<String>(), source.display());
+    let encoded_hash = hash.encode_hex::<String>();
+    let checksum_file_content = format!(
+        "{encoded_hash} {}\n",
+        source
+            .file_name()
+            .and_then(|name| name.to_str())
```

Comment:
I quite liked Aatif's suggestion:
```suggestion
            .and_then(std::ffi::OsStr::to_str)
```
:)